client/jetpack-connect/remote-credentials.js

```diff
@@ -77,13 +77,40 @@ export class OrgCredentialsForm extends Component {
 	}
 
 	componentDidUpdate() {
-		const { isResponseCompleted, siteToConnect } = this.props;
+		const { isResponseCompleted } = this.props;
 
 		if ( isResponseCompleted ) {
-			externalRedirect( addCalypsoEnvQueryArg( siteToConnect + REMOTE_PATH_AUTH ) );
+			// Login to remote site and redirect to JP connect URL
+			this.buildRemoteSiteLoginForm().submit();
 		}
 	}
 
+	buildRemoteSiteLoginForm() {
+		const { siteToConnect } = this.props;
+
+		const form = document.createElement( 'form' );
+		form.setAttribute( 'method', 'post' );
+
+		const redirectUrl = addCalypsoEnvQueryArg( REMOTE_PATH_AUTH );
+		const actionUrl = addQueryArgs( { redirect_to: redirectUrl }, siteToConnect + '/wp-login.php' );
```

Comment:
The hardcoded use of `/wp-login.php` worries me slightly.

In the endpoint, we attempt to scrape a login URL before falling back to `/wp-login.php`, but I don't know if the scraping ever works.

If is comes to it we could return a login URL from the endpoint.

Reply:
> we attempt to scrape a login URL before falling back to /wp-login.php, but I don't know if the scraping ever works.

I have not seen it at work for any of the sites yet.

Reply:
Noting here also that this form submission will not follow any redirects, so if user supplied an http url but site is https, the login will not work.

That is a moot point at the moment because the remote install does not follow redirects either, but we will need to use the correct URL here once we have determined it. I'm starting to think that the site-info endpoint (that we call before remote-install) should return the canonical URL.

Reply:
> if user supplied an http url but site is https, the login will not work.

This will be fixed by #23934